### PR TITLE
Wazo 3356 cannot edit contact

### PR DESCRIPTION
**Pull request by DrPyser**
https://wazo-dev.atlassian.net/browse/WAZO-3356

### CODE REVIEW

File: wazo_ui/helpers/classful.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Copyright 2018-2023 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
```
Comment (pc-m) [inline wazo_ui/helpers/classful.py:3]: a mypy check should be added to make sure added types are ok

**Review by pc-m — CHANGES_REQUESTED** [review]
When I visit `Phonebooks->Configuration` the list is empty.

File: wazo_ui/helpers/classful.py (hunk 1 above)
Comment (DrPyser) [reply wazo_ui/helpers/classful.py:3]: Agreed, but making this codebase palatable for mypy is non-trivial work which I would consider out-of-scope.
The way I see it, the added types should be useful for reviewing and understanding the code.

Comment (DrPyser) [reply wazo_ui/helpers/classful.py:3]: Though I admit I was a bit sloppy and overspecified some interfaces. Much of the generic code ends up being overriden and so the generic interface expectations don't matter that much.

Comment (DrPyser) [reply wazo_ui/helpers/classful.py:3]: Pushed commits that got the code to pass loose mypy checks:
```
mypy --allow-untyped-globals --allow-redefinition --ignore-missing-imports wazo_ui
wazo_ui/plugins/phonebook/service.py:13: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Success: no issues found in 264 source files
```
Now I could include this in tox directly, but the chosen way forward is to redirect tox to a `pre-commit` invocation (adding a `.pre-commit-config.yml` file for linters).

**Comment by DrPyser** [comment]
> When I visit `Phonebooks->Configuration` the list is empty.

Not sure I understand. The list of phonebooks? Did you create any?

**Comment by pc-m** [comment]
> > When I visit `Phonebooks->Configuration` the list is empty.
> 
> Not sure I understand. The list of phonebooks? Did you create any?

Yes, I have two phonebooks. The drop down list in Phonebooks->contacts contains my phonebooks but the Phonebooks->configuration list is empty.

Here's the result of the GET from wazo-dird
```
2023-09-15 13:44:33,892 [20186] (DEBUG) (wazo-dird): response body: {"items": [{"id": 2, "uuid": "f4e25a4a-8681-4211-8a3e-bac2e9e1de07", "name": "Mon Phonebook", "description": "Le test de phonebook", "tenant_uuid": "2c34c282-433e-4bb8-8d56-fec14ff7e1e9"}, {"id": 1, "uuid": "0997b615-95cd-453a-a69d-faacfa87107d", "name": "wazophonebook", "description": null, "tenant_uuid": "2c34c282-433e-4bb8-8d56-fec14ff7e1e9"}], "total": 2}
```

**Review by pc-m — APPROVED** [review]

**Comment by pc-m** [comment]
recheck

**Comment by pc-m** [comment]
recheck

**Pull request merged**